landa/water_body_management/doctype/catch_log_entry/catch_log_entry.js

```diff
@@ -2,9 +2,29 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on("Catch Log Entry", {
-	onload: function (frm) {
+	onload: (frm) => {
 		if (frm.is_new() && !frm.doc.year) {
 			frm.set_value("year", moment().year() - 1);
 		}
 	},
+	refresh: (frm) => {
+		if (frm.doc.docstatus < 1 && frm.doc.__unsaved)	{
+			// to avoid primary action refresh on dirty form (setting field value)
+			frm.disable_save();
+
+			frm.page.set_primary_action(__("Save and New"), () => {
+				frappe.run_serially([
+					() => frm.save(),
+					() => frm.reload_doc(),
+					() => frappe.new_doc("Catch Log Entry", {
+						organization: frm.doc.organization
+					}),
```

Comment:
```suggestion
					}),
					() => frm.refresh(),
```

Without a refresh the buttons for saving in the new form are not shown to me. 🤔

Reply:
I cleared cache, did a bench build, frappe and erpnext are on version-13-hotfix. Tried switching branch > clearing cache, rebuilding > switching back branch as well. Cannot replicate buttons not being there 😮
![2023-02-07 12 21 52](https://user-images.githubusercontent.com/25857446/217169655-7213aa8b-e246-47bf-911d-d5ffc50a349e.gif)

Reply:
@marination looks like your bench is too fast and we're facing a race condition on our slow ones:

https://user-images.githubusercontent.com/14891507/217608117-4f6005b3-df54-40ea-9a94-1e3cfc8d160d.mov

Reply:
Why is @marination having a faster bench? I fell like a second class dev. 😭

Reply:
Probably running on HAL 9000 (https://www.youtube.com/watch?v=njos57IJf-0)

Reply:
The issue seems to be role based. Frappe tries to render workflow actions as the primary action that's racing with the custom script.🙈

Reply:
Now it's working for admin but not for a "real" user 😄.

I'm afraid we'll have to solve the core issue: frappe loading data for the old form while we're already on a new one. Let me check with the client if it's worth the effort.